### ISSUE TITLE
LSQUnivariateSpline fitting failed with knots generated from UnivariateSpline

### ISSUE THREAD

**Issue by Fmajor** [scipy/scipy#5916]
`scipy.__version__='0.17.0'`

I tried to use the knots generated from UnivariateSpline to do a LSQUnivariateSpline fitting like this

```
fit = scipy.interpolate.UnivariateSpline( x, y, w=w, s=s)
knots = fit.get_knots()
newFit = scipy.interpolate.LSQUnivariateSpline(x, y, knots, w=w)
```

and got error
```

/usr/local/lib/python2.7/site-packages/scipy/interpolate/fitpack2.pyc in __init__(self, x, y, t, w, bbox, k, ext, check_finite)
    726         n = len(t)
    727         if not alltrue(t[k+1:n-k]-t[k:n-k-1] > 0, axis=0):
--> 728             raise ValueError('Interior knots t must satisfy '
    729                              'Schoenberg-Whitney conditions')
    730         if not dfitpack.fpchec(x, t, k) == 0:

ValueError: Interior knots t must satisfy Schoenberg-Whitney conditions
```

i have checked the knots=fit.get_knots(), the first and last value is the first and last value of x, but
line 727 means that the first and last knot CAN NOT be the first and last x data in LSQUnivariateSpline.

change 
`727         if not alltrue(t[k+1:n-k]-t[k:n-k-1] > 0, axis=0):`
to
`727         if not alltrue(t[k+1:n-k]-t[k:n-k-1] >= 0, axis=0):`

will solve this issue, but i'm not sure if this could cause other problems.

Any way, you can not init another fitting with the parameters got from a successful fitting, that's unreasonable...

**Comment by ev-br** [comment]
TL;DR: as a workaround, you likely want `newFit = LSQUnivariateSpline(x, y, knots[1:-1], w=w)`

Consider:

```
>>> x = np.arange(10)
>>> s = UnivariateSpline(x, x, s=0)
>>> s.get_knots()
array([ 0.,  2.,  3.,  4.,  5.,  6.,  7.,  9.])
>>> knt = s.get_knots()
>>> s1 = LSQUnivariateSpline(x, x, knt[1:-1])    # Chop 1st and last knot
>>> s1.get_knots()
array([ 0.,  2.,  3.,  4.,  5.,  6.,  7.,  9.])
```

Changing `knt[1:-1]` back to `knt` above gives the error you got. Walking a bit in pdb shows that

- internally, the knot vector is padded by boundary knots,

- `get_knots()` method chops up `k` boundary knots from either end, and

- `LSQUnivariateSpline` constructor adds `k+1` knots to each end.


This is a mess, but I do not see a way of fixing it in a backwards-compatible way.

**Comment by Fmajor** [comment]
ok, i have changed my program to use knots[1:-1] now

i think you can add your demo into the documents of LSQUnivariateSpline to make it more clear, for backwards-compatible purpose, just leave the code as it is.


i think this issue can be closed.

**Comment by ev-br** [comment]
If you have an idea of how to improve the docs, PRs are welcome!

**Comment by pv** [comment]
Adding an example how to do the above correctly to the LSQUnivariateSpline docstring maybe helps.

**Comment by Fmajor** [comment]
I agree with you.

**Comment by pv** [comment]
added in gh-5921
